Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/fragments/EmulationFragment.java

```diff
@@ -307,8 +316,28 @@ public synchronized void pause()
 			}
 		}
 
-		public synchronized void run()
+		public synchronized void run(boolean isActivityRecreated)
 		{
+			if (isActivityRecreated)
+			{
+				if (NativeLibrary.IsRunning())
+				{
+					loadPreviousTemporaryState = false;
+					state = State.PAUSED;
+					deleteFile(temporaryStatePath);
+				}
+				else
+				{
+					loadPreviousTemporaryState = true;
```

Comment:
It won't work isFinishing is also true when you rotate the device or press home button, so you will loose the state all the time. I tried to delete the saved state immediately after booting the game but that fails, I think it is not blocking so when the time comes to read the state it was gone, can we somehow block it similar to what you did for SaveStateAS?

Reply:
It would be possible to block, but I think it would be much simpler to let the core delete the savestate file once it's done loading it. I've now pushed a commit for that to my branch.

Reply:
That would be great! I will apply the changes soon.